### PR TITLE
in literal/comment stuff fixed and cleaned-up

### PR DESCRIPTION
**Pull request by andreas-roehler**
Might need your help still WRT rebase. Below the output of my endeavours so far - resorted to pull...

git rebase https://github.com/ocaml/tuareg.git
fatal: Needed a single revision
ungültiger Upstream-Branch https://github.com/ocaml/tuareg.git
ME$ git pull https://github.com/ocaml/tuareg.git
remote: Counting objects: 14, done.  
remote: Compressing objects: 100% (14/14), done.  
remote: Total 14 (delta 5), reused 0 (delta 0), pack-reused 0  
Unpacking objects: 100% (14/14), done.
Von https://github.com/ocaml/tuareg
- branch            HEAD       -> FETCH_HEAD
  automatischer Merge von tuareg.el
  KONFLIKT (Inhalt): Merge-Konflikt in tuareg.el
  Automatischer Merge fehlgeschlagen; beheben Sie die Konflikte und committen Sie dann das Ergebnis.opam/

### CODE REVIEW

**Comment by monnier** [comment]
> +  (let* ((pps (parse-partial-sexp (point-min) (point)))

That's a non-starter.  This should use (syntax-ppss).  And then only if
tuareg-use-syntax-ppss is non-nil.
Same for pretty much all other uses of parse-partial-sexp.



        Stefan

**Comment by andreas-roehler** [comment]
Am 10.06.2015 um 14:46 schrieb monnier:
> > + (let* ((pps (parse-partial-sexp (point-min) (point)))
>
> That's a non-starter. This should use (syntax-ppss).

Which calls parse-partial-sexp under the hood.

**Comment by Chris00** [comment]
On Wed, 10 Jun 2015 03:34:50 -0700, Andreas Röhler wrote:
> 
> Might your help still WRT rebase. Below the output of my endeavours so far -
> resorted to pull...
> 
> git rebase https://github.com/ocaml/tuareg.git
> fatal: Needed a single revision
> ungültiger Upstream-Branch https://github.com/ocaml/tuareg.git
> ME$ git pull https://github.com/ocaml/tuareg.git
> remote: Counting objects: 14, done.

If you did a `git clone https://github.com/ocaml/tuareg.git`, then the original repository is under the `origin` branch.  After a `git fetch` to make it up to date, just do a `git rebase origin` from your local branch.


> Unpacking objects: 100% (14/14), done.
> Von https://github.com/ocaml/tuareg
> 
>   * branch HEAD -> FETCH_HEAD automatischer Merge von tuareg.el KONFLIKT
>     (Inhalt): Merge-Konflikt in tuareg.el Automatischer Merge fehlgeschlagen;
>     beheben Sie die Konflikte und committen Sie dann das Ergebnis.opam/

Please set your locale to English if you want some help, I can't read
German messages.

**Comment by andreas-roehler** [comment]
Am 10.06.2015 um 15:30 schrieb Christophe Troestler:
> On Wed, 10 Jun 2015 03:34:50 -0700, Andreas Röhler wrote:
> >
> > Might your help still WRT rebase. Below the output of my endeavours 
> so far -
> > resorted to pull...
> >
> > git rebase https://github.com/ocaml/tuareg.git
> > fatal: Needed a single revision
> > ungültiger Upstream-Branch https://github.com/ocaml/tuareg.git
> > ME$ git pull https://github.com/ocaml/tuareg.git
> > remote: Counting objects: 14, done.
>
> If you did a `git clone https://github.com/ocaml/tuareg.git`, then the 
> original repository is under the `origin` branch. After a `git fetch` 
> to make it up to date, just do a `git rebase origin` from your local 
> branch.
>
>
Cloned at github and worked from my cloned repo.
"fetch" might be the word.

Will make tuareg ready for me to work with and re-start that process 
afterwards, if needed.


> > Unpacking objects: 100% (14/14), done.
> > Von https://github.com/ocaml/tuareg
> >
> > * branch HEAD -> FETCH_HEAD automatischer Merge von tuareg.el KONFLIKT
> > (Inhalt): Merge-Konflikt in tuareg.el Automatischer Merge 
> fehlgeschlagen;
> > beheben Sie die Konflikte und committen Sie dann das Ergebnis.opam/
>
> Please set your locale to English if you want some help, I can't read
> German messages.

Sorry, will consider this.

**Comment by monnier** [comment]
>> > + (let* ((pps (parse-partial-sexp (point-min) (point)))
>> That's a non-starter. This should use (syntax-ppss).
> Which calls parse-partial-sexp under the hood.

It does, but it's not the only thing it does and it doesn't call it in
the same way as you do.


        Stefan

**Comment by andreas-roehler** [comment]
Am 10.06.2015 um 17:28 schrieb monnier:
> >> > + (let* ((pps (parse-partial-sexp (point-min) (point)))
> >> That's a non-starter. This should use (syntax-ppss).
> > Which calls parse-partial-sexp under the hood.
>
> It does, but it's not the only thing it does and it doesn't call it in
> the same way as you do.
>
>
> Stefan

Its about broken and redundant code here. If you give an example where 
syntax-ppss does a better job then now working and XEmacs compatible 
parse-partial-sexp , I'll interested to see.

**Comment by monnier** [comment]
> Its about broken and redundant code here.

Redundant is because of backward compatibility.  I'd be happy to drop
support for Emacsen that don't have syntax-ppss, but that's not my call
to make.
We also have additional redundancy because we currently support two
indentation codebases.  Hopefully the non-SMIE indentation should
disappear at some point, but it's not my call either.

W.r.t broken, you still haven't given us any kind of description of an
actual concrete problem.

> If you give an example where
> syntax-ppss does a better job

syntax-ppss is all about caching the result of previous calls to
parse-partial-sexp so as to work better in very large buffer.
In 99% of the cases, the cache is filled as a side-effect of
font-locking, so syntax-ppss is super-fast in most cases even in
multi-MB buffers.  Using (parse-partial-sexp (point-min) (point)) in
your code is akin to using bubblesort: there has to be a damn good
reason to do so since it makes the code unusable on large buffers.

Also, as mentioned, syntax-ppss does more, since it also makes sure
syntax-propertize has been applied (currently, I think that's the only
significant difference).


        Stefan


PS: Of course, I'm biased against your code in any case:  I think that
Tuareg should aim to be integrated into GNU ELPA, and integrating code
from you is going against this goal since you're one of the rare few who
refuses to sign the needed paperwork.

File: .gitignore [hunk 1]
```diff
@@ -3,3 +3,7 @@
 *.tar.gz
 /tuareg.*/
 /opam/url
+commit.txt
+RCS/
+fehler.txt
+.gitignore
```
Comment (Chris00) [inline .gitignore]: `.gitignore` should not be ignored!  And there should be a newline.

File: .gitignore [hunk 2]
```diff
@@ -3,3 +3,7 @@
 *.tar.gz
 /tuareg.*/
 /opam/url
+commit.txt
+RCS/
+fehler.txt
```
Comment (Chris00) [inline .gitignore]: What are these 3 files good for?

File: tuareg-intern.el [hunk 3]
```diff
@@ -0,0 +1,119 @@
+;;; tuareg-intern.el --- Helper functions
```
Comment (Chris00) [inline tuareg-intern.el]: You put too many things in a single PR.  There are tests above, now this is a refactoring of the library.  Please split your PR in small easily reviewable chunks.

**Comment by Chris00** [comment]
> Of course, I'm biased against your code in any case: I think that Tuareg should aim to be integrated into GNU ELPA, and integrating code from you is going against this goal since you're one of the rare few who refuses to sign the needed paperwork.

@andreas-roehler Would you care to explain?  I would prefer not to do things that would prevent Tuareg to be able to (possibly) be integrated in GNU ELPA.  Is there a problem with signing the required papers if the need arises?

**Comment by andreas-roehler** [comment]
Am 21.06.2015 um 23:31 schrieb Christophe Troestler:
>
>     Of course, I'm biased against your code in any case: I think that
>     Tuareg should aim to be integrated into GNU ELPA, and integrating
>     code from you is going against this goal since you're one of the
>     rare few who refuses to sign the needed paperwork.
>
> @andreas-roehler <https://github.com/andreas-roehler> Would you care 
> to explain? I would prefer not to do things that would prevent Tuareg 
> to be able to (possibly) be integrated in GNU ELPA. Is there a problem 
> with signing the required papers if the need arises?
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/ocaml/tuareg/pull/54#issuecomment-113962182>.
>

Signed this years ago:

I'd like to ask you to sign a disclaimer for the program, thus putting
it in the public domain.  (For a small program such as this, it's not
worth our trying to assert a copyleft.)

     I, <name of person>, hereby disclaim all copyright interest in my
     program <program> which does <one-line description>.

     I affirm that I have no other intellectual property interest
     that would undermine this release, and will do nothing to undermine
     it in the future.  I represent that the work is my own and not
     a copy of someone else's work.

     <signature and date>

File: .gitignore (hunk 2 above)
Comment (andreas-roehler) [inline .gitignore]: Am 21.06.2015 um 23:24 schrieb Christophe Troestler:
>
> In .gitignore 
> <https://github.com/ocaml/tuareg/pull/54#discussion_r32897800>:
>
> > @@ -3,3 +3,7 @@
> >  *.tar.gz
> >  /tuareg.*/
> >  /opam/url
> > +commit.txt
> > +RCS/
> > +fehler.txt
>
> What are these 3 files good for?
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/ocaml/tuareg/pull/54/files#r32897800>.
>

Please edit these minor changes as you need it.

File: tuareg-intern.el (hunk 3 above)
Comment (andreas-roehler) [inline tuareg-intern.el]: Am 21.06.2015 um 23:28 schrieb Christophe Troestler:
>
> In tuareg-intern.el 
> <https://github.com/ocaml/tuareg/pull/54#discussion_r32897882>:
>
> > @@ -0,0 +1,119 @@
> > +;;; tuareg-intern.el --- Helper functions
>
> You put too many things in a single PR. There are tests above, now 
> this is a refactoring of the library. Please split your PR in small 
> easily reviewable chunks.
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/ocaml/tuareg/pull/54/files#r32897882>.
>

Hmm, don't know how to do that, as it's a translation from python-mode.el

The API needs some helper functions, which by themselfs wouldn't be that 
impressive and probably never get merged.

Basics is in commit 2ed2643dd6c48d6de8f0e99d55efd4

with *644 additions* and *8 deletions*.

Maybe run the tests to get some impression?

**Pull request closed without merging**